Found local copy...
Loading...
Done!
[I 2025-09-06 01:25:44,727] A new study created in RDB with name: __init__
[I 2025-09-06 01:26:53,570] A new study created in RDB with name: MOL2VEC_KNN_PAMPA_NCATS_0
[I 2025-09-06 01:28:30,020] A new study created in RDB with name: MOL2VEC_KNN_PAMPA_NCATS_1
[I 2025-09-06 01:30:40,180] A new study created in RDB with name: MOL2VEC_KNN_PAMPA_NCATS_8
[I 2025-09-06 01:30:53,960] A new study created in RDB with name: MOL2VEC_KNN_PAMPA_NCATS_9
[I 2025-09-06 01:30:54,035] A new study created in RDB with name: MOL2VEC_KNN_PAMPA_NCATS_7
[I 2025-09-06 01:30:54,206] A new study created in RDB with name: MOL2VEC_KNN_PAMPA_NCATS_4
[I 2025-09-06 01:32:46,212] Trial 0 finished with value: 0.4530397057533264 and parameters: {'n_neighbors': 11, 'weights': 'uniform', 'metric': 'chebyshev'}. Best is trial 0 with value: 0.4530397057533264.
[I 2025-09-06 01:32:46,232] Trial 0 finished with value: 0.4578434228897095 and parameters: {'n_neighbors': 7, 'weights': 'distance', 'metric': 'minkowski'}. Best is trial 0 with value: 0.4578434228897095.
[I 2025-09-06 01:32:46,243] Trial 0 finished with value: 0.4554089307785034 and parameters: {'n_neighbors': 5, 'weights': 'uniform', 'metric': 'chebyshev'}. Best is trial 0 with value: 0.4554089307785034.
[I 2025-09-06 01:32:46,736] Trial 1 finished with value: 0.4554912745952606 and parameters: {'n_neighbors': 3, 'weights': 'uniform', 'metric': 'minkowski'}. Best is trial 0 with value: 0.4554089307785034.
[I 2025-09-06 01:32:47,699] Trial 1 finished with value: 0.45957738161087036 and parameters: {'n_neighbors': 9, 'weights': 'distance', 'metric': 'manhattan'}. Best is trial 0 with value: 0.4578434228897095.
[I 2025-09-06 01:32:47,733] Trial 1 finished with value: 0.4518963098526001 and parameters: {'n_neighbors': 11, 'weights': 'uniform', 'metric': 'manhattan'}. Best is trial 1 with value: 0.4518963098526001.
[I 2025-09-06 01:32:48,318] Trial 2 finished with value: 0.4550415575504303 and parameters: {'n_neighbors': 9, 'weights': 'distance', 'metric': 'manhattan'}. Best is trial 2 with value: 0.4550415575504303.
[I 2025-09-06 01:32:48,773] Trial 3 finished with value: 0.45512738823890686 and parameters: {'n_neighbors': 9, 'weights': 'distance', 'metric': 'minkowski'}. Best is trial 2 with value: 0.4550415575504303.
[I 2025-09-06 01:32:49,009] Trial 2 finished with value: 0.4581730365753174 and parameters: {'n_neighbors': 5, 'weights': 'distance', 'metric': 'manhattan'}. Best is trial 0 with value: 0.4578434228897095.
[I 2025-09-06 01:32:49,027] Trial 2 finished with value: 0.4493862986564636 and parameters: {'n_neighbors': 7, 'weights': 'distance', 'metric': 'chebyshev'}. Best is trial 2 with value: 0.4493862986564636.
[I 2025-09-06 01:32:50,084] Trial 4 finished with value: 0.45593422651290894 and parameters: {'n_neighbors': 3, 'weights': 'uniform', 'metric': 'manhattan'}. Best is trial 2 with value: 0.4550415575504303.
[I 2025-09-06 01:32:50,650] Trial 3 finished with value: 0.45938852429389954 and parameters: {'n_neighbors': 3, 'weights': 'uniform', 'metric': 'manhattan'}. Best is trial 0 with value: 0.4578434228897095.
[I 2025-09-06 01:32:50,672] Trial 3 finished with value: 0.45192036032676697 and parameters: {'n_neighbors': 9, 'weights': 'uniform', 'metric': 'manhattan'}. Best is trial 2 with value: 0.4493862986564636.
[I 2025-09-06 01:32:51,301] Trial 4 finished with value: 0.4608444571495056 and parameters: {'n_neighbors': 9, 'weights': 'uniform', 'metric': 'minkowski'}. Best is trial 0 with value: 0.4578434228897095.
[I 2025-09-06 01:32:51,320] Trial 4 finished with value: 0.4504025876522064 and parameters: {'n_neighbors': 3, 'weights': 'uniform', 'metric': 'minkowski'}. Best is trial 2 with value: 0.4493862986564636.
[I 2025-09-06 01:32:51,476] Trial 5 finished with value: 0.4549006521701813 and parameters: {'n_neighbors': 3, 'weights': 'distance', 'metric': 'manhattan'}. Best is trial 5 with value: 0.4549006521701813.
[I 2025-09-06 01:32:52,675] Trial 5 finished with value: 0.45917224884033203 and parameters: {'n_neighbors': 9, 'weights': 'distance', 'metric': 'chebyshev'}. Best is trial 0 with value: 0.4578434228897095.
[I 2025-09-06 01:32:52,696] Trial 5 finished with value: 0.4528920352458954 and parameters: {'n_neighbors': 9, 'weights': 'uniform', 'metric': 'chebyshev'}. Best is trial 2 with value: 0.4493862986564636.
[I 2025-09-06 01:32:53,028] Trial 6 finished with value: 0.4573524296283722 and parameters: {'n_neighbors': 7, 'weights': 'uniform', 'metric': 'manhattan'}. Best is trial 5 with value: 0.4549006521701813.
[I 2025-09-06 01:32:53,255] Trial 6 finished with value: 0.45680636167526245 and parameters: {'n_neighbors': 3, 'weights': 'distance', 'metric': 'minkowski'}. Best is trial 6 with value: 0.45680636167526245.
[I 2025-09-06 01:32:53,442] Trial 7 finished with value: 0.45512738823890686 and parameters: {'n_neighbors': 9, 'weights': 'distance', 'metric': 'minkowski'}. Best is trial 5 with value: 0.4549006521701813.
[I 2025-09-06 01:32:53,843] Trial 6 finished with value: 0.4484385848045349 and parameters: {'n_neighbors': 5, 'weights': 'distance', 'metric': 'chebyshev'}. Best is trial 6 with value: 0.4484385848045349.
[I 2025-09-06 01:32:53,924] Trial 7 finished with value: 0.4608444571495056 and parameters: {'n_neighbors': 9, 'weights': 'uniform', 'metric': 'minkowski'}. Best is trial 6 with value: 0.45680636167526245.
[I 2025-09-06 01:32:53,987] Trial 8 finished with value: 0.45314615964889526 and parameters: {'n_neighbors': 5, 'weights': 'distance', 'metric': 'minkowski'}. Best is trial 8 with value: 0.45314615964889526.
[I 2025-09-06 01:32:54,114] Trial 7 finished with value: 0.44999071955680847 and parameters: {'n_neighbors': 11, 'weights': 'distance', 'metric': 'minkowski'}. Best is trial 6 with value: 0.4484385848045349.
[I 2025-09-06 01:32:54,241] Trial 8 finished with value: 0.45680636167526245 and parameters: {'n_neighbors': 3, 'weights': 'distance', 'metric': 'minkowski'}. Best is trial 6 with value: 0.45680636167526245.
[I 2025-09-06 01:32:55,441] Trial 9 finished with value: 0.45332807302474976 and parameters: {'n_neighbors': 3, 'weights': 'uniform', 'metric': 'chebyshev'}. Best is trial 8 with value: 0.45314615964889526.
[I 2025-09-06 01:32:55,636] Trial 8 finished with value: 0.45174863934516907 and parameters: {'n_neighbors': 11, 'weights': 'distance', 'metric': 'chebyshev'}. Best is trial 6 with value: 0.4484385848045349.
[I 2025-09-06 01:32:55,695] Trial 9 finished with value: 0.45972511172294617 and parameters: {'n_neighbors': 7, 'weights': 'uniform', 'metric': 'manhattan'}. Best is trial 6 with value: 0.45680636167526245.
[I 2025-09-06 01:32:55,870] Trial 10 finished with value: 0.45314615964889526 and parameters: {'n_neighbors': 5, 'weights': 'distance', 'metric': 'minkowski'}. Best is trial 8 with value: 0.45314615964889526.
[I 2025-09-06 01:32:56,487] Trial 11 finished with value: 0.45314615964889526 and parameters: {'n_neighbors': 5, 'weights': 'distance', 'metric': 'minkowski'}. Best is trial 8 with value: 0.45314615964889526.
[I 2025-09-06 01:32:56,830] Trial 9 finished with value: 0.45174863934516907 and parameters: {'n_neighbors': 11, 'weights': 'distance', 'metric': 'chebyshev'}. Best is trial 6 with value: 0.4484385848045349.
[I 2025-09-06 01:32:56,870] Trial 12 finished with value: 0.45525097846984863 and parameters: {'n_neighbors': 11, 'weights': 'distance', 'metric': 'minkowski'}. Best is trial 8 with value: 0.45314615964889526.
[I 2025-09-06 01:32:57,139] Trial 10 finished with value: 0.4609302878379822 and parameters: {'n_neighbors': 11, 'weights': 'distance', 'metric': 'chebyshev'}. Best is trial 6 with value: 0.45680636167526245.
[I 2025-09-06 01:32:57,171] Trial 10 finished with value: 0.4464092552661896 and parameters: {'n_neighbors': 5, 'weights': 'distance', 'metric': 'minkowski'}. Best is trial 10 with value: 0.4464092552661896.
[I 2025-09-06 01:32:57,377] Trial 13 finished with value: 0.45314615964889526 and parameters: {'n_neighbors': 5, 'weights': 'distance', 'metric': 'minkowski'}. Best is trial 8 with value: 0.45314615964889526.
[I 2025-09-06 01:32:57,496] Trial 11 finished with value: 0.4464092552661896 and parameters: {'n_neighbors': 5, 'weights': 'distance', 'metric': 'minkowski'}. Best is trial 10 with value: 0.4464092552661896.
[I 2025-09-06 01:32:57,649] Trial 11 finished with value: 0.45680636167526245 and parameters: {'n_neighbors': 3, 'weights': 'distance', 'metric': 'minkowski'}. Best is trial 6 with value: 0.45680636167526245.
[I 2025-09-06 01:32:57,669] Trial 14 finished with value: 0.45314615964889526 and parameters: {'n_neighbors': 5, 'weights': 'distance', 'metric': 'minkowski'}. Best is trial 8 with value: 0.45314615964889526.
[I 2025-09-06 01:32:57,935] Trial 12 finished with value: 0.45680636167526245 and parameters: {'n_neighbors': 3, 'weights': 'distance', 'metric': 'minkowski'}. Best is trial 6 with value: 0.45680636167526245.
[I 2025-09-06 01:32:57,956] Trial 12 finished with value: 0.4464092552661896 and parameters: {'n_neighbors': 5, 'weights': 'distance', 'metric': 'minkowski'}. Best is trial 10 with value: 0.4464092552661896.
[I 2025-09-06 01:32:58,126] Trial 13 finished with value: 0.45680636167526245 and parameters: {'n_neighbors': 3, 'weights': 'distance', 'metric': 'minkowski'}. Best is trial 6 with value: 0.45680636167526245.
[I 2025-09-06 01:33:02,153] Trial 13 finished with value: 0.4464092552661896 and parameters: {'n_neighbors': 5, 'weights': 'distance', 'metric': 'minkowski'}. Best is trial 10 with value: 0.4464092552661896.
[I 2025-09-06 01:33:07,364] Trial 14 finished with value: 0.4464092552661896 and parameters: {'n_neighbors': 5, 'weights': 'distance', 'metric': 'minkowski'}. Best is trial 10 with value: 0.4464092552661896.
[I 2025-09-06 01:33:37,999] Trial 14 finished with value: 0.45680636167526245 and parameters: {'n_neighbors': 3, 'weights': 'distance', 'metric': 'minkowski'}. Best is trial 6 with value: 0.45680636167526245.
concurrent.futures.process._RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/home/ucnwdma/miniconda3/envs/internship/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1819, in _execute_context
    self.dialect.do_execute(
  File "/home/ucnwdma/miniconda3/envs/internship/lib/python3.9/site-packages/sqlalchemy/engine/default.py", line 732, in do_execute
    cursor.execute(statement, parameters)
sqlite3.OperationalError: database is locked

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/ucnwdma/miniconda3/envs/internship/lib/python3.9/site-packages/optuna/storages/_rdb/storage.py", line 78, in _create_scoped_session
    session.commit()
  File "/home/ucnwdma/miniconda3/envs/internship/lib/python3.9/site-packages/sqlalchemy/orm/session.py", line 1435, in commit
    self._transaction.commit(_to_root=self.future)
  File "/home/ucnwdma/miniconda3/envs/internship/lib/python3.9/site-packages/sqlalchemy/orm/session.py", line 829, in commit
    self._prepare_impl()
  File "/home/ucnwdma/miniconda3/envs/internship/lib/python3.9/site-packages/sqlalchemy/orm/session.py", line 808, in _prepare_impl
    self.session.flush()
  File "/home/ucnwdma/miniconda3/envs/internship/lib/python3.9/site-packages/sqlalchemy/orm/session.py", line 3367, in flush
    self._flush(objects)
  File "/home/ucnwdma/miniconda3/envs/internship/lib/python3.9/site-packages/sqlalchemy/orm/session.py", line 3507, in _flush
    transaction.rollback(_capture_exception=True)
  File "/home/ucnwdma/miniconda3/envs/internship/lib/python3.9/site-packages/sqlalchemy/util/langhelpers.py", line 70, in __exit__
    compat.raise_(
  File "/home/ucnwdma/miniconda3/envs/internship/lib/python3.9/site-packages/sqlalchemy/util/compat.py", line 207, in raise_
    raise exception
  File "/home/ucnwdma/miniconda3/envs/internship/lib/python3.9/site-packages/sqlalchemy/orm/session.py", line 3467, in _flush
    flush_context.execute()
  File "/home/ucnwdma/miniconda3/envs/internship/lib/python3.9/site-packages/sqlalchemy/orm/unitofwork.py", line 456, in execute
    rec.execute(self)
  File "/home/ucnwdma/miniconda3/envs/internship/lib/python3.9/site-packages/sqlalchemy/orm/unitofwork.py", line 630, in execute
    util.preloaded.orm_persistence.save_obj(
  File "/home/ucnwdma/miniconda3/envs/internship/lib/python3.9/site-packages/sqlalchemy/orm/persistence.py", line 245, in save_obj
    _emit_insert_statements(
  File "/home/ucnwdma/miniconda3/envs/internship/lib/python3.9/site-packages/sqlalchemy/orm/persistence.py", line 1238, in _emit_insert_statements
    result = connection._execute_20(
  File "/home/ucnwdma/miniconda3/envs/internship/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1631, in _execute_20
    return meth(self, args_10style, kwargs_10style, execution_options)
  File "/home/ucnwdma/miniconda3/envs/internship/lib/python3.9/site-packages/sqlalchemy/sql/elements.py", line 325, in _execute_on_connection
    return connection._execute_clauseelement(
  File "/home/ucnwdma/miniconda3/envs/internship/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1498, in _execute_clauseelement
    ret = self._execute_context(
  File "/home/ucnwdma/miniconda3/envs/internship/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1862, in _execute_context
    self._handle_dbapi_exception(
  File "/home/ucnwdma/miniconda3/envs/internship/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 2043, in _handle_dbapi_exception
    util.raise_(
  File "/home/ucnwdma/miniconda3/envs/internship/lib/python3.9/site-packages/sqlalchemy/util/compat.py", line 207, in raise_
    raise exception
  File "/home/ucnwdma/miniconda3/envs/internship/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1819, in _execute_context
    self.dialect.do_execute(
  File "/home/ucnwdma/miniconda3/envs/internship/lib/python3.9/site-packages/sqlalchemy/engine/default.py", line 732, in do_execute
    cursor.execute(statement, parameters)
sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) database is locked
[SQL: INSERT INTO studies (study_name) VALUES (?)]
[parameters: ('MOL2VEC_KNN_PAMPA_NCATS_3',)]
(Background on this error at: https://sqlalche.me/e/14/e3q8)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/ucnwdma/miniconda3/envs/internship/lib/python3.9/concurrent/futures/process.py", line 243, in _process_worker
    r = call_item.fn(*call_item.args, **call_item.kwargs)
  File "/myriadfs/home/ucnwdma/Scratch/UCL_internship/code/study_manager.py", line 97, in run_single_experiment
    best_hyperparams = self.run_hyperparameter_optimization(
  File "/myriadfs/home/ucnwdma/Scratch/UCL_internship/code/study_manager.py", line 73, in run_hyperparameter_optimization
    study = optuna.create_study(
  File "/home/ucnwdma/miniconda3/envs/internship/lib/python3.9/site-packages/optuna/_convert_positional_args.py", line 135, in converter_wrapper
    return func(**kwargs)  # type: ignore[call-arg]
  File "/home/ucnwdma/miniconda3/envs/internship/lib/python3.9/site-packages/optuna/study/study.py", line 1299, in create_study
    study_id = storage.create_new_study(direction_objects, study_name)
  File "/home/ucnwdma/miniconda3/envs/internship/lib/python3.9/site-packages/optuna/storages/_cached_storage.py", line 78, in create_new_study
    study_id = self._backend.create_new_study(directions=directions, study_name=study_name)
  File "/home/ucnwdma/miniconda3/envs/internship/lib/python3.9/site-packages/optuna/storages/_rdb/storage.py", line 282, in create_new_study
    session.add(models.StudyModel(study_name=study_name, directions=direction_models))
  File "/home/ucnwdma/miniconda3/envs/internship/lib/python3.9/contextlib.py", line 126, in __exit__
    next(self.gen)
  File "/home/ucnwdma/miniconda3/envs/internship/lib/python3.9/site-packages/optuna/storages/_rdb/storage.py", line 95, in _create_scoped_session
    raise optuna.exceptions.StorageInternalError(message) from e
optuna.exceptions.StorageInternalError: An exception is raised during the commit. This typically happens due to invalid data in the commit, e.g. exceeding max length. 
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/myriadfs/home/ucnwdma/Scratch/UCL_internship/./code/main.py", line 18, in <module>
    manager.run_nested_cv(fingerprint, model_name, dataset_name)
  File "/myriadfs/home/ucnwdma/Scratch/UCL_internship/code/study_manager.py", line 136, in run_nested_cv
    raise exc
  File "/myriadfs/home/ucnwdma/Scratch/UCL_internship/code/study_manager.py", line 131, in run_nested_cv
    seed_result, test_predictions, test_indices = future.result()
  File "/home/ucnwdma/miniconda3/envs/internship/lib/python3.9/concurrent/futures/_base.py", line 438, in result
    return self.__get_result()
  File "/home/ucnwdma/miniconda3/envs/internship/lib/python3.9/concurrent/futures/_base.py", line 390, in __get_result
    raise self._exception
optuna.exceptions.StorageInternalError: An exception is raised during the commit. This typically happens due to invalid data in the commit, e.g. exceeding max length. 
